var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -265,14 +265,16 @@ class Cp2k(MakefilePackage, CudaPackage, CMakePackage, ROCmPackage):
     # like ELPA, SCALAPACK are independent and Spack will ensure that
     # a consistent/compatible combination is pulled into the dependency graph.
     with when("+sirius"):
-        depends_on("sirius+fortran+shared")
+        depends_on("sirius+fortran")
+        depends_on("sirius+scalapack", when="+mpi")
         depends_on("sirius+cuda", when="+cuda")
         depends_on("sirius+rocm", when="+rocm")
         depends_on("sirius+openmp", when="+openmp")
         depends_on("sirius~openmp", when="~openmp")
         depends_on("sirius@7.0.0:7.0", when="@8:8.2")
         depends_on("sirius@7.2", when="@8.3:8.9")
         depends_on("sirius@7.3:", when="@9.1")
+        depends_on("sirius@7.3.2", when="@2023.1")
```

Comment:
Unfortunately, I haven't tested other versions. However, since the version of Sirius specified in cp2k-2023.1/arch/*psmp is v7.3.2, I believe specifying 7.3.2 is the safest choice.